samples/snippets/src/main/java/IdTokenFromImpersonatedCredentials.java

```diff
@@ -0,0 +1,159 @@
+ */
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken.Payload;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdTokenVerifier;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.json.gson.GsonFactory;
+import com.google.api.services.iamcredentials.v1.IAMCredentials;
+import com.google.api.services.iamcredentials.v1.IAMCredentials.Projects.ServiceAccounts.GenerateIdToken;
+import com.google.api.services.iamcredentials.v1.model.GenerateIdTokenRequest;
+import com.google.auth.http.HttpCredentialsAdapter;
+import com.google.auth.oauth2.GoogleCredentials;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public class IdTokenFromImpersonatedCredentials {
+
+  public static void main(String[] args) throws IOException, GeneralSecurityException {
+    // TODO(Developer): Replace the below variables before running the code.
+
+    // Your Google Cloud project id.
+    String projectId = "your-google-cloud-project-id";
+
+    // Path to the service account json credential file.
+    String jsonCredentialPath = "path-to-json-credential-file";
+
+    // Provide the scopes that you might need to request to access Google APIs,
+    // depending on the level of access you need.
+    // Example: The following scope lets you view and manage Pub/Sub topics and subscriptions.
+    // For more information, see: https://developers.google.com/identity/protocols/oauth2/scopes
+    String scope = "https://www.googleapis.com/auth/pubsub";
+
+    // The service name for which the id token is requested. Service name refers to the
+    // logical identifier of an API service, such as "pubsub.googleapis.com".
+    String targetAudience = "pubsub.googleapis.com";
+
+    // The service account name of the limited-privilege account for whom the credential is created.
+    String impersonatedServiceAccount = "name@project.service.gserviceaccount.com";
+
+    getIdTokenFromImpersonatedCredentials(
+        projectId, jsonCredentialPath, impersonatedServiceAccount, scope, targetAudience);
+  }
+
+  public static void getIdTokenFromImpersonatedCredentials(
+      String projectId,
+      String jsonCredentialPath,
+      String impersonatedServiceAccount,
+      String scope,
+      String targetAudience)
+      throws GeneralSecurityException, IOException {
+
+    // Initialize the IAMCredentials service with the source credential and scope.
+    IAMCredentials service = null;
+    try {
+      service = initService(jsonCredentialPath, scope);
+    } catch (IOException | GeneralSecurityException e) {
+      System.out.println("Unable to initialize service: \n" + e);
+      return;
+    }
+
+    // delegates: The chained list of delegates required to grant the final accessToken.
```

Comment:
I've also separated out the delegation flow from the single service account flow. It doesn't make sense to create an ID token from a delegation chain, because there's no authz included for ID tokens. So let's just stick to a single privilege-bearing (current term, subject to change) service account (no delegation chain).

Reply:
The delegates parameter is mandatory in the "Impersonated Credentials" API, unlike the other one we used from IAM library. 
Hence, I am unable to remove it. Added clarifying comments stating that it's optional. wdyt?

Reply:
Could we just say 
// delegates: Unused for this call
or something like that? I'm concerned that the subject of chaining is very complex, and that folks will spend a lot of time trying to parse and understand what identities need to have what permission on what-- only to get to the bottom and find out it's unused. Why put them through that?